test/functional-portable/kubernetes/noncloud/deploymenttemplate_test.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes_test
+
+import (
+	"context"
+	"encoding/json"
+	"path"
+	"testing"
+	"time"
+
+	aztoken "github.com/radius-project/radius/pkg/azure/tokencredentials"
+	"github.com/radius-project/radius/pkg/cli/clients_new/generated"
+	radappiov1alpha3 "github.com/radius-project/radius/pkg/controller/api/radapp.io/v1alpha3"
+	"github.com/radius-project/radius/pkg/controller/reconciler"
+	"github.com/radius-project/radius/pkg/sdk"
+	sdkclients "github.com/radius-project/radius/pkg/sdk/clients"
+	"github.com/radius-project/radius/test/rp"
+	"github.com/radius-project/radius/test/testcontext"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+	controller_runtime "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func Test_DeploymentTemplate(t *testing.T) {
+	defaultProviderConfig, err := generateDefaultProviderConfig()
+	require.NoError(t, err)
+
+	testcases := []struct {
+		name             string
+		namespace        string
+		fileName         string
+		templateFilePath string
+		providerConfig   string
+		parameters       map[string]string
+	}{
+		{
+			name:             "dt-env",
+			namespace:        "dt-ns-env",
+			fileName:         "env.bicep",
+			templateFilePath: path.Join("testdata", "env", "env.json"),
+			providerConfig:   defaultProviderConfig,
+			parameters: map[string]string{
+				"name":      "dt-env",
+				"namespace": "dt-ns-env",
+			},
+		},
+		{
+			name:             "dt-module",
+			namespace:        "dt-ns-module",
+			fileName:         "module.bicep",
+			templateFilePath: path.Join("testdata", "module", "module.json"),
+			providerConfig:   defaultProviderConfig,
+			parameters: map[string]string{
+				"name":      "dt-module",
+				"namespace": "dt-ns-module",
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := testcontext.New(t)
+			opts := rp.NewRPTestOptions(t)
+
+			name := tc.name
+			namespace := tc.namespace
+
+			template, err := afero.ReadFile(afero.NewOsFs(), tc.templateFilePath)
```

Comment:
I'm not a huge fan of how viral `afero` is becoming in this PR. It's pretty easy to use `testdata` for inputs in tests, and temp directories for outputs.

Reply:
I removed `afero` and added a small filesystem abstraction that we can use instead.